api-spec/extensions/fields/README.md

```diff
@@ -2,32 +2,68 @@
-
+By default, the STAC search endpoint `/search` returns all attributes of each Item, as there is no way to specify 
+exactly those attributes that should be returned. The Fields API Extension adds new functionality that allows the 
+client to suggest to the server which Item attributes should be included or excluded in the response. 
+
+When calling `/search` using POST with`Content-Type: application/json`, this extension adds an attribute `fields` with 
+an object value to the core JSON search request body. The `fields` object contains two attributes with string array 
+values, `include` and `exclude`.
+
+When calling `/search` using GET or POST with `Content-Type: application/x-www-form-urlencoded` or 
+`Content-Type: multipart/form-data`, the semantics are the same, except the syntax is a single parameter `fields` with 
+a comma-separated list of attribute names, where `exclude` values are those prefixed by a `-` and `include` values are 
+those with no prefix, e.g., `-geometry`, or `id,-geometry,properties`.
+
+It is recommended that implementations provide exactly the `include` and `exclude` sets specified by the request, 
+but this is not required. These values are only hints to the server as to the desires of the client, and not a 
+contract about what the response will be. Implementations are still considered compliant if fields not specified as part of `include` 
+are in the response or ones specified as part of `exclude` are.  For example, implementations may choose to always 
+include simple string fields like `id` and `type` regardless of the `exclude` specification. However, it is recommended 
+that implementations honor excludes for attributes with more complex and arbitrarily large values 
+(e.g., `geometry`, `assets`).  For example, some Items may have a geometry with a simple 5 point polygon, but these 
+polygons can be very large when reprojected to EPSG:4326, as in the case of a highly-decimated sinusoidal polygons.
+Implementations are also not required to implement semantics for nested values whereby one can include an object, but
+exclude attributes of that object, e.g., include `properties` but exclude `properties.datetime`.
+
+No error should be returned if a specified field has no value for it in the catalog.  For example, if the attribute 
+"properties.eo:cloud_cover" is specified but there is no cloud cover value for an Item or the API does not even 
+support the EO Extension, a successful HTTP response should be returned and the Item entities will not contain that 
+attribute. 
+
+If no `fields` are specified, the response is **must** be a valid ItemCollection. If a client excludes
+attributes that are required in a STAC Item, the server may return an invalid STAC Item. For example, if `type` 
+and `geometry` are excluded, the entity will not even be a valid GeoJSON Feature, or if `bbox` is excluded then the entity 
```

Comment:
Just a (stupid?) thought, but maybe it would be better to set some(?) fields to null instead of removing them. A null geometry would be valid GeoJSON, but removing it would be invalid GeoJSON. Depends on how important it is for us to return valid GeoJSON in most cases, but still allow removing large fields... Same applies for STAC Item compliance. An empty object would be compliant with the spec, but removing would not be compliant. Similar things apply for (most?) objects/arrays, I guess.

Reply:
It looks like while technically it is a GeoJSON with a null geometry, most clients won't be able to handle it and generate an error.
GeoJSON must also contain `properties`, even if empty, and also must contain the `type` field, so a minimal FeatureCollection would look like:
```
{
   "type": "FeatureCollection"
   "features": [
      {
         "type": "Feature",
         "geometry": null,
         "properties": {}
      }
   ]
}
```

It's really not a bad idea to keep our response as valid GeoJSON, otherwise using the fields query will sometimes be valid and sometimes not.

Reply:
I feel the problem with this is that the result should never contain non-nested attributes with  values that are not the actual values of the Item.

Reply:
If we restrict to type, geometry and properties, I think it would be fine.

Reply:
I also do not have a strong opinion here.  It seemed like it might be a good idea to ensure responses are always valid, but I'm also fine with saying "if you use fields the response may not be valid GeoJSON".